x-pack/plugins/apm/public/components/routing/templates/apm_service_template.tsx

```diff
@@ -100,6 +114,43 @@ function Template({
   );
 }
 
+function AnalyzeDataButton({ serviceName }: { serviceName: string }) {
+  const { agentName } = useApmServiceContext();
+  const { services } = useKibana();
+  const { urlParams } = useUrlParams();
+  const { rangeTo, rangeFrom, environment } = urlParams;
+  const basepath = services.http?.basePath.get();
+
+  if (isRumAgentName(agentName) || isIosAgent(agentName)) {
+    const href = createExploratoryViewUrl(
+      {
+        'apm-series': {
+          dataType: isRumAgentName(agentName) ? 'ux' : 'mobile',
+          time: { from: rangeFrom, to: rangeTo },
+          reportType: 'kpi-over-time',
+          reportDefinitions: {
+            'service.name': [serviceName],
+            'service.environment': [environment],
```

Comment:
i think if environment is ENVIRONMENT_ALL, you should skip passing environment. 
otherwise exploratory view will consider that as a term and not return anything.